group/assignment6/RearrrangingCars.java

```diff
@@ -0,0 +1,71 @@
+package assignment_6;
+
+import java.util.Hashtable;
+import java.util.LinkedList;
+import java.util.Map.Entry;
+import java.util.Random;
+
+public class RearrrangingCars {
+
+	private Hashtable<Integer, Character> start;
+	private Hashtable<Integer, Character> end;
+	private Random rand;
+	private int noCarKey = 0;
+	private int noExistKey = -1;
+
+	public RearrrangingCars(Hashtable<Integer, Character> start,
+			Hashtable<Integer, Character> end) {
+		rand = new Random();
+		this.start = start;
+		this.end = end;
+	}
+	
+	/**
+	 * given an starting and ending state of parking
+	 * start with the parking emptySlot in start state,
+	 * get the car that will exist at the same slot in the end state 
+	 * and do a move by swap the car from its current slot to the emptySlot
+	 * repeat that until reaching emptySlots
+	 * finally check if the start state equal the end,
+	 * if not it will be the case that at the beginning the emptySlot at the same position in two states
+	 * in this case generate in the first move a random slot to swap it with the empty 
+	 * and then starting the algorithm
+	 * @return linkedList of sequences to define moves as car, starting and ending spaces
+	 */
+	public LinkedList<Sequence> generateMoves() {
+		LinkedList<Sequence> moves = new LinkedList<>();
+		char emptySlot = start.get(noCarKey);
+		move(moves, noCarKey, emptySlot);
+		if (!start.equals(end)) {

```

Comment:
shouldn't this here be a `while`  instead of `if` ? Because I think that currently it wouldn't detect the following scenario:
`start = [1,2,3,4,5,0]` and `end = [2,1,4,3,0,5]` 
it would do something like:
``[1,2,3,4,0,5] // checks once and sees that start != end so it continues to ``
``[2,1,3,4,0,5] // but doesn't further check although 3 and 4 still need swapping ``

Reply:
i change the algorithm to handle this case and cases similar to this 
i won't check after the whole process ends. i will check in every time i generate carID if its empty and we didn't get the right answer yet then we generate another one random
i think this will fix that bug